scipy/_build_utils/_generate_blas_wrapper.py

```diff
@@ -79,11 +79,15 @@ def generate_file_wrapper(sigs, accelerate):
     return ''.join(file_text)
 
 def make_all(outdir,
-             blas_signature_file=os.path.join(
-                 LINALG_DIR, "cython_blas_signatures.txt"),
-             lapack_signature_file=os.path.join(
-                 LINALG_DIR, "cython_lapack_signatures.txt"),
+             blas_signature_file=None,
+             lapack_signature_file=None,
              accelerate=False):
+    blas_signature_file = os.path.join(
+                 LINALG_DIR, "cython_blas_signatures.txt"
+    ) if blas_signature_file is None else blas_signature_file
+    lapack_signature_file = os.path.join(
+                 LINALG_DIR, "cython_lapack_signatures.txt"
+    ) if lapack_signature_file is None else lapack_signature_file
```

Comment:
These changes don't seem needed. `os.path.join` produces a string, and strings are immutable. 

If it's because of a linter false positive then of course I don't mind the change. It's more idiomatic and readable to write it as:
```python
if blas_signature_file is None:
    blas_signature_file = ...
```
though, so please change to that if a change is needed.

Reply:
Ah, this is the same thing @fancidev referred to - it's present in many places, just here it's affecting readability more because the keyword name is so long.

I agree with @fancidev's preference for the multi-line version in general. For short lines I won't comment on it though, since both flavors are pretty readable.

Reply:
>I agree with fancidev's preference for the multi-line version in general. For short lines I won't comment on it though, since both flavors are pretty readable.

Thanks, fixed in 37da874a97eda46a3c36f2a242a8c5746a46f0c5.